pkg/tools/etcd_helper.go

```diff
@@ -469,7 +471,16 @@ func (h *EtcdHelper) SetObj(key string, obj, out runtime.Object, ttl uint64) err
 
 // Pass an EtcdUpdateFunc to EtcdHelper.GuaranteedUpdate to make an etcd update that is guaranteed to succeed.
 // See the comment for GuaranteedUpdate for more detail.
-type EtcdUpdateFunc func(input runtime.Object) (output runtime.Object, ttl uint64, err error)
+type EtcdUpdateFunc func(input runtime.Object, node *etcd.Node, res *etcd.Response) (output runtime.Object, ttl *uint64, err error)
```

Comment:
I'd strongly prefer not to expose etcd.Node outside of this package, because I'd like to evolve this into a generic storage interface. Can we provide only object & ttl as input?

(arguably the proper way to do this is to encode ttl into our objects, in metadata or as an annotation-- then none of these interfaces would need to change, but that's much more work of course)